python/pyspark/sql/utils.py

```diff
@@ -307,3 +311,52 @@ def is_timestamp_ntz_preferred() -> bool:
     """
     jvm = SparkContext._jvm
     return jvm is not None and jvm.PythonSQLUtils.isTimestampNTZPreferred()
+
+
+def is_remote() -> bool:
+    """
+    Returns if the current running environment is for Spark Connect.
+    """
+    return "SPARK_REMOTE" in os.environ
+
+
+def try_remote_functions(f: FuncT) -> FuncT:
```

Comment:
I don't know it well, is it possible to do monkey patching instead of add this annotation to each function?

Reply:
It cannot do because the import can be done first before Spark session initialization. We are only able to know if we're using the remove or not when Spark session is initialized. e.g.,

```python
from pyspark.sql.functions import lit
from pyspark.sql import SparkSession

SparkSession.builder.getOrCreate()  # we can know if this is remote or not from here.
```

Reply:
got it, thank for the explanation